### PR TITLE
Core packages and separating native example for users from control-app for internal testing purposes

### PR DESCRIPTION
**Pull request by Xiltyn**
### Summary

This PR emerges as a proposition and a start point for a conversation about the structure and clarity of our repository. The discussion arose after conversation about our current Next.js example PR.

It is currently very confusing in terms of readability which parts of the monorepo serve what purpose. It would be a logical confusion for a contributor to expect all `/packages` to be exported, even though `-logger` is only for internal use. It would also make sense to think that since `examples/native` is exported as a workspace a new example would also need to be exported as such. 

Because we require a testing environment to assure our internal stability as well as be able to live test changes in the library or changes in react-native (main focus atm), I propose the following ::

1. We separate the examples, I created a new RN example and called it `native-cli` as opposed to `native-expo` and following the naming convention set in react-native docs.
2. I moved current `examples/native` package into new directory called `/core` to signify core internal packages (name is WIP ;) ). I also changed its name to `reassure-core-control-app`.
3. In order to keep with the convention of having export API modules separately, I moved `reassure-logger` to the `/core` directory and renamed it to `reassure-core-logger`.
4. I removed `examples/native` from our workspaces and put `core/*` there instead. I also made sure they are built in first place in case dependency order matters here, I'm not sure myself.
5. I updated all scripts and pipelines and everything around `example/native` in order to reflect the new structure properly.

### Test plan

Currently the example is passing on local. They can be run using the `yarn run test:control-app` command from the root directory.

If the pipeline properly outputs the results everything should be fine since it will also point to the new directory.

### Future plans

We would like to eventually handle the `examples` in some different manner, potentially a separate repository as per my discussion with @mdjastrzebski. In which case I deem this to be a fair first step towards that goal. It also improves upon the readability of the monorepo and will in turn help future contributors navigate this space. 

Wdyt? @mdjastrzebski @thymikee

### CODE REVIEW

**Comment by Xiltyn** [comment]
I apologise for the **unholly** number of files changes by moving the examples about and creating new ones does it...

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -27,16 +27,16 @@ jobs:
         run: cd ./docusaurus && yarn install --force
 
       - name: Validate TypeScript
-        run: yarn typescript && yarn typescript:example
+        run: yarn typescript && yarn typescript:control-app
 
       - name: Validate ESLint
-        run: yarn lint && yarn lint:example
```
Comment (Xiltyn) [inline .github/workflows/main.yml:33]: I would like to mention that this is either a bug or obsolete as far as I can understand. Running eslint checks all folders so running it again from a child folder in structure just runs another eslint for that directory only. 

Please check the pattern we're using for eslint in the attached screenshot

Screenshot 2023-03-09 at 17 07 45

**Comment by Xiltyn** [comment]
I'll work on aligning the pipelines, they are being wonky because of the previous configs, but it's a matter of time. Anyway, discussion is the point of this after all, so let's focus on this and when we're done, the pipelines will be done as well.

**Comment by mdjastrzebski** [comment]
I'm not sure whether relatively small gain of clarifying the `examples/native` status is worth such large scale changes. If you want to continue on this endeavour I would recommend following course of action:

1. PR 1: Move `examples/native` to `test-apps/native`. Updated all paths, etc. No other code changes, no extracting loggers, etc. That PR will be large but easy to review.

2. PR 2: Generate a new `examples/native-cli` using fresh RN CLI. There is no need to sticking to old code here. Copy relevant config and test files. No other code changes, no extracting of loggers, etc. That PR will also be large, but primarily will be a CLI generated stuff, where we could drop the noisy `android` & `ios` folders, so that would be also relatively easy to review.

3. In case you want to do any other code changes, pls create separate PRs for them, each with clearly defined goal.

Re naming: `reassure-core-control-app` sounds kinda mysterious. Since we are using this app for actual testing, I would recommend simply `test-apps/native` or similar folder name.

@Xiltyn wdyt?

**Comment by mdjastrzebski** [comment]
Replaced by #392 #391

**Pull request closed without merging**